tests/foreman/ui/test_classparameters.py

```diff
@@ -74,11 +172,22 @@ def setUpClass(cls):
                 'per_page': 1000
             })
 
-    @classmethod
-    def tearDownClass(cls):
-        """Removes puppet class."""
-        super(SmartClassParametersTestCase, cls).tearDownClass()
-        delete_puppet_class(cls.puppet_class.name)
+        cls.host = entities.Host(organization=cls.session_org).create()
+        cls.host.environment = cls.env
+        cls.host.update(['environment'])
+        cls.host.add_puppetclass(data={'puppetclass_id': cls.puppet_class.id})
+        cls.domain_name = entities.Domain(id=cls.host.domain.id).read().name
+
+    # TearDown brakes parallel tests run as every test depends on the same
```

Comment:
Not sure it is needed to comment out `tearDownClass` as this TestCase anyway should be run in one thread.